File: scripts/query-rls-policies.ts
#!/usr/bin/env npx tsx
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';

dotenv.config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseServiceKey = process.env.SUPABASE_SERVICE_ROLE_KEY!;

async function queryRLSPolicies() {
  const supabase = createClient(supabaseUrl, supabaseServiceKey);

  console.log('🔍 Querying RLS Policies on jobs and users_extended tables...\n');

  // Query policies via direct SQL
  const { data, error } = await supabase.rpc('exec_sql', {
    sql: `
      SELECT 
        schemaname,
        tablename,
        policyname,
        permissive,
        roles,
        cmd,
        qual,
        with_check
      FROM pg_policies
      WHERE tablename IN ('jobs', 'users_extended')
      ORDER BY tablename, policyname;
    `
  });

  if (error) {
    console.error('❌ Error querying policies:', error);
    return;
  }

  console.log('📋 RLS Policies Found:\n');
  console.log(JSON.stringify(data, null, 2));
}

queryRLSPolicies().catch(console.error);

File: query_rls_policies.py
#!/usr/bin/env python3
import requests
import json

# From .env.local
SUPABASE_URL = "https://rtwigjwqufozqfwozpvo.supabase.co"
SUPABASE_SERVICE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InJ0d2lnandxdWZvenFmd296cHZvIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1NDI1MDMwMCwiZXhwIjoyMDY5ODI2MzAwfQ.e4U3aDv5GDIFiPlY_JcveGwbAT9p-ahiW_0hhoOUoY0"

headers = {
    "apikey": SUPABASE_SERVICE_KEY,
    "Authorization": f"Bearer {SUPABASE_SERVICE_KEY}",
    "Content-Type": "application/json"
}

# Query to get all RLS policies
sql_query = """
SELECT
    schemaname,
    tablename,
    policyname,
    cmd,
    qual,
    with_check
FROM pg_policies
WHERE schemaname = 'public'
ORDER BY tablename, policyname;
"""

# Execute SQL query using PostgREST
response = requests.post(
    f"{SUPABASE_URL}/rest/v1/rpc/exec_sql",
    headers=headers,
    json={"query": sql_query}
)

print(f"Status Code: {response.status_code}")
print(f"Response: {response.text}")

# If the exec_sql function doesn't exist, try direct query
if response.status_code != 200:
    print("\n\nTrying direct query approach...")
    # Try querying pg_policies directly
    response = requests.get(
        f"{SUPABASE_URL}/rest/v1/pg_policies?schemaname=eq.public&select=*&order=tablename,policyname",
        headers=headers
    )
    print(f"Status Code: {response.status_code}")
    print(f"Response: {response.text}")
